ecmascript/transforms/src/typescript/strip.rs

```diff
@@ -680,12 +713,22 @@ impl VisitMut for Strip {
                     .scope
                     .imported_idents
                     .get(&(local.sym.clone(), local.span.ctxt()));
-                match entry {
-                    Some(&DeclInfo {
-                        has_type: true,
-                        has_concrete: false,
-                    }) => false,
-                    _ => true,
+                match self.config.import_not_used_as_values {
+                    ImportNotUsedAsValues::Remove => match entry {
+                        Some(&DeclInfo {
+                            has_concrete: false,
+                            ..
+                        }) => false,
+                        _ => true,
+                    },
+                    ImportNotUsedAsValues::Preserve => match entry {
+                        Some(&DeclInfo {
+                            has_type: true,
+                            has_concrete: false,
+                            ..
+                        }) => false,
+                        _ => true,
+                    },
```

Comment:
@kdy1 This is not the behaviour of `tsc` under `"importsNotUsedAsValues": "preserve"`. The correct behaviour is to _never_ preserve completely unused import identifiers. The above branch should always be used.

I think `ImportNotUsedAsValues` should be removed since it's a misunderstanding of the equivalent `tsc` option.

Reply:
I'll make a breaking change to fix it.